### PR TITLE
Add PW88 keyboard

### PR DESCRIPTION
**Pull request by SmollChungus**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/chibios
Comment (drashna) [inline lib/chibios:1]: You have changes to the chibios submoudle, which I assume you didn't mean to do.

This should be simpler to fix than in the past. Specifically: 
```sh
cd lib/chibios
git checkout qmk-develop
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: lib/chibios
Comment (SmollChungus) [reply lib/chibios:1]: Definitely unintentional, my apologies.

About the proposed fix above, I'm not certain how to exclude the chibios in this current PR. Would it be easier to start a new pr?

Comment (drashna) [reply lib/chibios:1]: the above commands should revert to the proper version.  Committing that and pushing it should fix the PR.

File: keyboards/pw88/config.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#define RGBLIGHT_EFFECT_SNAKE
+#define RGBLIGHT_EFFECT_KNIGHT
+#define RGBLIGHT_EFFECT_CHRISTMAS
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_EFFECT_TWINKLE
+#define RGB_VAL_STEP 12
+
+#define RGBLIGHT_LAYERS
```
Comment (waffle87) [inline keyboards/pw88/config.h:41]:
```suggestion
```

File: keyboards/pw88/info.json [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        "pid": "0x8888",
+        "device_version": "0.0.1"
+    },
+    "matrix_pins": {
+        "cols": ["GP4", "GP3", "GP2", "GP26", "GP25", "GP24", "GP23", "GP20", "GP19", "GP18", "GP29", "GP28", "GP27", "GP14", "GP1", "GP5", "GP6"],
+        "rows": ["GP8", "GP11", "GP10", "GP9", "GP13", "GP12"]
+    },
+    "diode_direction": "COL2ROW",
+    "processor": "RP2040",
+    "bootloader": "rp2040",
```
Comment (waffle87) [inline keyboards/pw88/info.json:17]:
```suggestion
    "bootloader": "rp2040",
    "rgblight": {
        "led_count": 34,
        "layers": {
             "enabled": true
        },
        "animation": {
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
        }
    },
```

File: keyboards/pw88/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  [3] = LAYOUT_all(
+    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,               KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,   KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,               KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,               KC_TRNS,
+    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    MO(2),                          KC_TRNS,
+    KC_TRNS,  KC_TRNS, KC_TRNS,                                    KC_TRNS,                                        KC_TRNS,    MO(1),    KC_TRNS,    KC_TRNS,               KC_TRNS, KC_TRNS, KC_TRNS
+  ),
+
```
Comment (waffle87) [inline keyboards/pw88/keymaps/default/keymap.c:55]:
```suggestion
    KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,   KC_SLSH,    KC_RSFT,    _______,                            KC_UP,
    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT,  MO(1),      KC_RGUI,    KC_RCTL,                 KC_LEFT, KC_DOWN, KC_RGHT
    ),

  [1] = LAYOUT_all(
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,               RGB_TOG, RGB_MOD, RGB_RMOD,
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,   KC_TRNS,    RGB_HUI, RGB_SAI, RGB_VAI,
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,               RGB_HUD, RGB_SAD, RGB_VAD,
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,               KC_TRNS,
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    _______,                          KC_VOLU,
    KC_TRNS,  KC_TRNS, KC_TRNS,                                    KC_MPLY,                                        KC_TRNS,    _______, _______,      KC_TRNS,               KC_MPRV, KC_VOLD, KC_MNXT
  )
```

File: keyboards/pw88/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+/*
+void keyboard_post_init_user(void) {
+    // Initialize RGB to static black
+    rgblight_enable_noeeprom();
+   rgblight_sethsv_noeeprom(HSV_AZURE);
+   rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
+}
+*/
```
Comment (waffle87) [inline keyboards/pw88/keymaps/default/keymap.c:66]:
```suggestion
```

File: keyboards/pw88/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,   KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,               KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,               KC_TRNS,
+    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    MO(2),                          KC_TRNS,
+    KC_TRNS,  KC_TRNS, KC_TRNS,                                    KC_TRNS,                                        KC_TRNS,    MO(1),    KC_TRNS,    KC_TRNS,               KC_TRNS, KC_TRNS, KC_TRNS
+  ),
+
+};
+
+
```
Comment (waffle87) [inline keyboards/pw88/keymaps/via/keymap.c:58]:
```suggestion
    KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,   KC_SLSH,    KC_RSFT,    _______,                            KC_UP,
    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT,  MO(1),      KC_RGUI,    KC_RCTL,                 KC_LEFT, KC_DOWN, KC_RGHT
    ),

  [1] = LAYOUT(
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,               RGB_TOG, RGB_MOD, RGB_RMOD,
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,   KC_TRNS,    RGB_HUI, RGB_SAI, RGB_VAI,
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,               RGB_HUD, RGB_SAD, RGB_VAD,
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,               KC_TRNS,
    KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,   KC_TRNS,    _______,                          KC_VOLU,
    KC_TRNS,  KC_TRNS, KC_TRNS,                                    KC_MPLY,                                        KC_TRNS,    _______, _______,      KC_TRNS,               KC_MPRV, KC_VOLD, KC_MNXT
  )
};
```

File: keyboards/pw88/readme.md
Comment (waffle87) [inline keyboards/pw88/readme.md:1]: This file need to follow the [template](https://github.com/qmk/qmk_firmware/blob/develop/data/templates/keyboard/readme.md?plain=1).

File: keyboards/pw88/rules.mk [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+WS2812_DRIVER = vendor      # RGB driver
```
Comment (waffle87) [inline keyboards/pw88/rules.mk:14]:
```suggestion
# This file intentionally blank
```

File: keyboards/pw88/info.json [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    "layouts": {
```
Comment (waffle87) [inline keyboards/pw88/info.json:18]:
```suggestion
    "ws2812": {
        "pin": "GP0",
        "driver": "vendor"
    }, 
    "layouts": {
```

File: keyboards/pw88/info.json [hunk 8]
```diff
@@ -0,0 +1,115 @@
+{
+    "keyboard_name": "PW88",
+    "manufacturer": "Smollchungus",
+    "url": "https://www.github.com/smollchungus",
+    "maintainer": "Smollchungus",
+    "usb": {
+        "vid": "0x5363",
+        "pid": "0x8888",
+        "device_version": "0.0.1"
+    },
```
Comment (waffle87) [inline keyboards/pw88/info.json:10]:
```suggestion
    },
    "features": {
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true,
        "rgblight": true,
        "nkro": true
    },
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: lib/chibios
Comment (SmollChungus) [reply lib/chibios:1]: Thank you for the instructions, I believe its fixed now, although I'm not completly sure how to check

**Review by SmollChungus — COMMENTED** [review]
i believe this covers everything

File: keyboards/pw88/readme.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
+# PW88
+
+![%PW88%](https://i.imgur.com/0uN5u2g.jpg)
```
Comment (drashna) [inline keyboards/pw88/readme.md:3]:
```suggestion
![%PW88%](https://i.imgur.com/0uN5u2gh.jpg)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**